### PR TITLE
test: setup e2e tests env for atm feature

### PR DESCRIPTION
**Pull request by zhiying-lin**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Enable atm feature for the shared-vnet networking settings

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] run `make reviewable` for basic local test
- [ ] Read and followed fleet-networking's [Code of conduct](https://github.com/Azure/fleet-networking/blob/main/CODE_OF_CONDUCT.md).
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

### How has this code been tested

https://github.com/Azure/fleet-networking/actions/runs/12111480140/job/33763409219

### Special notes for your reviewer

we can support atm feature for the other networking settings in a separate PR

### CODE REVIEW

File: test/scripts/bootstrap.sh [hunk 1]
```diff
@@ -173,7 +300,8 @@ kubectl apply -f config/crd/*
 helm install hub-net-controller-manager \
     ./charts/hub-net-controller-manager/ \
     --set image.repository=$REGISTRY/hub-net-controller-manager \
-    --set image.tag=$TAG
+    --set image.tag=$TAG \
+        $( [ "$ENABLE_TRAFFIC_MANAGER" = "true" ] && echo "--set enableTrafficManagerFeature=true -f hub_azure_config.yaml" )
```
Comment (jwtty) [inline test/scripts/bootstrap.sh:304]: I didn't know this would work.

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -37,6 +37,13 @@ jobs:
       fail-fast: false
       matrix:
         network-setting: [shared-vnet, peered-vnet,  dynamic-ip-allocation]
+        include:
+          - network-setting: shared-vnet
+            enable-traffic-manager: true
```
Comment (jwtty) [inline .github/workflows/e2e-tests.yml:42]: Why do we only enable traffic manager in shared-vnet scenario?

File: test/scripts/aks-shared-vnet.sh [hunk 3]
```diff
@@ -37,7 +37,8 @@ az network vnet subnet create \
     --address-prefixes 10.2.0.0/16
 
 # Create aks member cluster1.
-az aks create \
+if [ "$ENABLE_TRAFFIC_MANAGER" == "false" ]; then
```
Comment (jwtty) [inline test/scripts/aks-shared-vnet.sh:40]: Can we just always create clusters with BYO identities?

File: test/scripts/bootstrap.sh (hunk 1 above)
Comment (zhiying-lin) [reply test/scripts/bootstrap.sh:304]: i learned this from you :)

File: .github/workflows/e2e-tests.yml (hunk 2 above)
Comment (zhiying-lin) [reply .github/workflows/e2e-tests.yml:42]: we will support others in a separate PR. i don't want to put too many changes in one PR.

File: test/scripts/aks-shared-vnet.sh (hunk 3 above)
Comment (zhiying-lin) [reply test/scripts/aks-shared-vnet.sh:40]: would like to keep one test for testing without ATM to make sure without any permission, mcs should work.

Comment (jwtty) [reply test/scripts/aks-shared-vnet.sh:40]: We can only assign permissions when enabling ATM but always use BYO identities?

Comment (ryanzhang-oss) [reply test/scripts/aks-shared-vnet.sh:40]: I think we need to keep no BYO identity clustrers so we know it still works

**Review by jwtty — APPROVED** [review]

**Pull request merged**